### PR TITLE
osbuilder: Allow image registry to be customizable

### PR DESCRIPTION
**Pull request by egernst**
Give the user chance to specify their own registry in event the default
provided are not accessible, desirable.

Fixes: #1393

Signed-off-by: Eric Ernst <eric.g.ernst@gmail.com>

### CODE REVIEW

File: tools/osbuilder/image-builder/image_builder.sh [hunk 1]
```diff
@@ -138,6 +141,7 @@ build_with_container() {
 	image_name=$(basename "${image}")
 
 	"${container_engine}" build  \
+		   --build-arg IMAGE_REGISTRY="${IMAGE_REGISTRY}" \
```
Comment (wainersm) [inline tools/osbuilder/image-builder/image_builder.sh:144]: Hi @egernst, wouldn't make sense to pass the argument only and only if IMAGE_REGISTRY is exported by the user? Otherwise it will always overwrite the default in Dockerfile.

Comment (egernst) [reply tools/osbuilder/image-builder/image_builder.sh:144]: Yeah, agreed, I was just looking at that. In that case, we'd want to drop the default at top of file, and then do something like:

```bash
               REGISTRY_ARG=""
+               if [ -n "${IMAGE_REGISTRY}" ]; then
+                       echo "it was set"
+                       REGISTRY_ARG="--build-arg IMAGE_REGISTRY=${IMAGE_REGISTRY}"
+               fi
+
                # setup to install go or rust here
                generate_dockerfile "${distro_config_dir}"
                "$container_engine" build  \
-                       --build-arg IMAGE_REGISTRY="${IMAGE_REGISTRY}" \
+                       ${REGISTRY_ARG} \
                        --build-arg http_proxy="${http_proxy}" \
                        --build-arg https_proxy="${https_proxy}" \
                        -t "${image_name}" "${distro_config_dir}"
 ```

**Review by wainersm — APPROVED** [review]
Looks good and work for me!

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 2]
```diff
@@ -126,13 +123,23 @@ GRACEFUL_EXIT       If set, and if the DISTRO configuration specifies a
...
 
 ROOTFS_DIR          Path to the directory that is populated with the rootfs.
                     Default value: <${script_name} path>/rootfs-<DISTRO-name>
 
+RUST_AGENT          When set to "no", build kata-agent from go agent instead of kata-rust-agent
+                    Default value: "yes"
+
+RUST_AGENT_PKG      URL of the Git repository hosting the agent package.
+                    Default value: ${RUST_AGENT_PKG}
+
```
Comment (fidencio) [inline tools/osbuilder/rootfs-builder/rootfs.sh:142]: I'd put those one if the DOCKER_RUNTIME re-shuffle as part of a different patch in, but in the same series.

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 3]
```diff
@@ -116,6 +110,9 @@ AGENT_SOURCE_BIN    Path to the directory of agent binary.
 DISTRO_REPO         Use host repositories to install guest packages.
                     Default value: <not set>
 
+DOCKER_RUNTIME      Docker runtime to use when USE_DOCKER is set.
+                    Default value: runc
+
```
Comment (fidencio) [inline tools/osbuilder/rootfs-builder/rootfs.sh:115]: You're just shuffling this one around, right?  As it's not exactly related to this patch, I'd put it in a different patch (but part of the same series).

**Review by fidencio — COMMENTED** [review]
@egernst, the idea seems fine to be, but I have a suggestion of leaving the parameters sort for a different patch (in the same series, in the same PR).

What do you think?  If you consider it too much of a nitpick, let me know.

**Comment by egernst** [comment]
Updated:
 - separate commit for alphabet fun
 - fixup image-builder script.
 
 PTAL @fidencio !

**Review by likebreath — APPROVED** [review]

**Review by fidencio — APPROVED** [review]
AGENT_VERSION should be after AGENT_SOURCE_BIN, I guess that's the only one missing.

Apart from that, lgtm!
Thanks, @egernst!
